soc/xtensa/intel_adsp/common/cavs_ipc.c

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) 2022 Intel Corporation
```

Comment:
why put the driver under the SOC and not under drivers/ipc/....

Reply:
It's a SOC-specific device for now.  I'll see about adding an extra API layer above it that can be more portably implemented

Reply:
ack - this needs to SOC and protocol generic so it can be used by any SOC. At at mid level we need something like (feel free to modify)
```c
int ipc_send_message(uint32_t addr, char *doorbell, size_t door_size, char *mailbox, size_t mail_size, int blocking);
int ipc_recv_message(uint32_t addr, int (callback_func*) (doorbell, door_size, mailbox, mailsize));
```
The higher level application API would in cSOF would be similar to what it is today, but it would do the protocol mapping and call the above zephyr APIs.

Reply:
Inbound in maybe an hour.  Really it just productizes the "simple" bits of this hardware (so no weird return/DONE interrupt) as a generic subsystem API and implements it with the device as it stands already.  Finishing the typing part, need to write a test and it'll go onto the end of this PR.